packages/common/container-definitions/src/loader.ts

```diff
@@ -389,21 +389,48 @@ export interface IContainer extends IEventProvider<IContainerEvents> {
 	proposeCodeDetails(codeDetails: IFluidCodeDetails): Promise<boolean>;
 
 	/**
-	 * Attaches the Container to the Container specified by the given Request.
+	 * When a container is created in Fluid it starts in what we call a detached state,
+	 * which means it is detached from server storage and unavailable for other users to open or collaborate with.
+	 * A detached container can be serialized, stored, and rehydrated by the application.
 	 *
-	 * @privateRemarks
+	 * To finish the creation of a container and make it available for other users to open and collaborate with it must be attached,
+	 * which means it is attached to or persisted in server storage. This function is what accomplishes that.
+	 *
+	 * In order to craft the IRequest parameter necessary to call this function you should use functionally provided
+	 * by the driver you are using. This is usually a free function called createCreateNewRequest, which should take
+	 * strongly typed arguments related to your driver and server.
+	 *
+	 * By default an attach failure will result in the closure of the container.
+	 * However, closure can now be avoided in most cased by setting:
+	 * Fluid.Container.RetryOnAttachFailure to true
+	 * via the config provider passed to the loader.
+	 *
+	 * In the event of this method throwing an exception, you should check if the container is closed via the closed property.
+	 * If the container is not closed, it is safe to continue editing the container, calling serialize to capture the container's state,
+	 * or try calling attach again.
 	 *
-	 * TODO - in the case of failure options should give a retry policy.
-	 * Or some continuation function that allows attachment to a secondary document.
 	 */
 	attach(
 		request: IRequest,
 		attachProps?: { deltaConnection?: "none" | "delayed" },
 	): Promise<void>;
 
 	/**
-	 * Extract a snapshot of the container as long as it is in detached state. Calling this on an attached container
-	 * is an error.
+	 * This function will capture the state of a container this is not Attached or closed.
+	 * This is useful in draft-like scenarios. For example, think of an email draft, where a user can make changes over time,
```

Comment:
Not so sure this comment is that related to serialize, it sounds more like detached to me.

Reply:
granted, serialize is only used in detached, attaching scenarios but in the phrase "This is useful" not sure if "this" is detached/attaching state or serialize. I guess it makes sense both ways but it was a bit confusing

Reply:
Agreed, I have a comment above re: attach() where I think this sort of explanation would fit better.

Reply:
all containers start detached, and all container needs to be attached. serialize is what allows preserving a unattached container, for draft like scenarios. i don't see why this wouldn't belong heere